content/6_Advanced/Count_Min.mdx

```diff
@@ -3,38 +3,219 @@ id: count-min
 redirects:
   - /adv/CntMin
 title: 'Counting Minimums with Segment Tree'
-author: Benjamin Qi
+author: Benjamin Qi, Dustin Miao
 prerequisites:
   - RURQ
-description: "?"
+description: "Querying for the minimum and number of occurences of minimum in a range"
 frequency: 0
 ---
 
-## Implementation
+<FocusProblem problem = "sam" />
 
-segment tree that keeps track of minimum and # of minimums
+<Resources>
+	<Resource
+		source="cp-algo"
+		title="Finding max and number of occurences"
+		url="data_structures/segment_tree.html#toc-tgt-8"
+		starred
+	/>
+</Resources>
 
-## Application - Rectangle Union
+We would like a data structure that can efficiently handle two types of operations:
+1. Update index $i$ to value $v$
+2. Report the minimum and the number of occurences of the minimum on a range $[l, r]$
 
-<FocusProblem problem="sam" />
+We can use a normal segment tree to handle range queries, but slightly modify each node and the merge operation. Let each node be a pair of values $(\texttt{val}, \texttt{cnt})$, where $\texttt{val}$ is the minimum value and $\texttt{cnt}$ is the number occurences of the minimum value. 
 
-<Spoiler title="Hint">
+If node $c$ has two children $a$ and $b$, then
+- if $a.\texttt{val} < b.\texttt{val}$, then $c = a$
+- if $a.\texttt{val} > b.\texttt{val}$, then $c = b$
+- if $a.\texttt{val} = b.\texttt{val}$, then $c = \{a.\texttt{val}, a.\texttt{cnt} + b.\texttt{cnt}\}$
 
-For each $x$-coordinate, count the number of squares that are covered by
-**none** of the input rectangles.
+### Implementation
+
+<LanguageSection>
+
+<CPPSection>
+
+```cpp
+const int MAXN = 2e5;
+const int INF = INT32_MAX;
+
+struct Node {
+	int val = INF, cnt = 1;
+} T[2 * MAXN];
```

Comment:
rename T to something else

Reply:
I think T is fine? It's what most people use